Cargo.toml

```diff
@@ -52,7 +52,7 @@ homepage = "https://datafusion.apache.org"
 license = "Apache-2.0"
 readme = "README.md"
 repository = "https://github.com/apache/datafusion"
-rust-version = "1.73"
+rust-version = "1.75"
```

Comment:
By my reading of https://github.com/apache/datafusion?tab=readme-ov-file#rust-version-compatibility-policy

is that since 1.74 was [released in Nov 2023](https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1740-2023-11-16), we can safely update in this PR